loader.js

```diff
@@ -141,15 +141,16 @@ async function sendData(port, data) {
       console.log('Booting...');
       await sendCommand(port, prepareCommand(commands.BOOT), false);
     } else {
-      const partial = Buffer.from(data.buffer, offset, STATUS_UPDATE_AT);
+      var bytesToWrite = Math.min(size-offset, STATUS_UPDATE_AT);
+      const partial = Buffer.from(data.buffer, offset, bytesToWrite);
 
       if (offset === MEG_32) {
```

Comment:
We may also need to adjust for 32 meg threshold as some file 32m + a few bytes will not get uploaded properly.
This was implmented with the assumption that it will be always a multiple of 1meg :)

Reply:
Threshold is unaffected, my bad.